subt/tools/submit.py

```diff
@@ -0,0 +1,155 @@
+)
+
+
+def _cmd(*args):
+    return subprocess.run(args, stdout=subprocess.PIPE).stdout
+
+
+def validate_world(world):
+    try:
+        return WORLDS[world]
+    except KeyError:
+        sys.exit(f"'{world}' not valid world identification")
+
+
+def validate_robots(robots, timeout):
+    valid = {}
+    for name, kind in robots.items():
+        try:
+            valid[name] = ROBOTS[kind]
+        except KeyError:
+            sys.exit(f"'{kind}' not valid robot kind identification")
+    valid[f"T{timeout}"] = "TEAMBASE"
+    return valid
+
+
+def validate_image(image):
+    image_bytes = bytes(image, 'ascii')
+    images = _cmd('aws', 'ecr', 'list-images', '--repository-name', 'subt/robotika')
+    for img in images.split(b'\n'):
+        parts = img.split(b'\t')
+        if parts[-1] == image_bytes:
+            return f'200670743174.dkr.ecr.us-east-1.amazonaws.com/subt/robotika:{image}'
+    sys.exit(f"image '{image}' not found in aws subt/robotika repository")
+
+
+def generate_simname(logdir, image, world):
+    if not logdir.exists():
+        sys.exit(f"{logdir} does not exist")
+    image = image.replace('-', '').replace('_', '')
+    prefix = image+world
+    existing = [x for x in logdir.iterdir() if x.is_dir() and x.name.startswith(prefix)]
+    run = 0
+    for a in sorted(existing):
+        run = re.search('(?<=[vr])[0-9]+$', a.name).group()
+    run = int(run) + 1
+    simname = f"{prefix}r{run}"
+    return simname
+
+
+def _pretty_print_request(req):
+    print('{}\n{}\r\n{}\r\n\r\n{}'.format(
+        '-----------START-----------',
+        req.method + ' ' + req.url,
+        '\r\n'.join('{}: {}'.format(k, v) for k, v in req.headers.items()),
+        req.body.decode('utf-8'),
+    ))
+
+
+def submit(token, simname, image, world, robots, dont_submit):
+    data = [
+        ('name', (None, simname)),
+        ('owner', (None, "robotika")),
+        ('circuit', (None, world)),
+    ]
+    for name, kind in robots.items():
+        data.append(('robot_name', (None, name)))
+        data.append(('robot_type', (None, kind)))
+        data.append(('robot_image', (None, image)))
+
+    headers = {
+        'Private-Token': token
+    }
+
+    req = requests.Request('POST', url="https://cloudsim.ignitionrobotics.org/1.0/simulations", files=data,
+                           headers=headers).prepare()
+
+    if dont_submit:
+        print("======= not really submiting =======")
+        _pretty_print_request(req)
+        return False
+
+    s = requests.Session()
+    resp = s.send(req)
+
+    print(json.dumps(resp.json(), indent=4))
+
+    if resp.status_code == 200:
+        return True
+
+    return False
+
+
+def main(argv):
+    token = os.getenv("SUBT_ACCESS_TOKEN")
+    if token is None:
+        sys.exit("SUBT_ACCESS_TOKEN environment variable not set")
+
+    parser = argparse.ArgumentParser(description='Submit cloudsim run')
+    parser.add_argument("config", nargs="?", default=str(pathlib.Path('./run.toml')))
```

Comment:
I am blind again :( ... sorry - so maybe please add some description or what is the default:
```
gpu@gpu-desktop:~/git/osgar/subt/docker$ python3 submit.py -h
usage: submit.py [-h] [-n] [config]

Submit cloudsim run

positional arguments:
  config

optional arguments:
  -h, --help  show this help message and exit
  -n          don't really submit
```